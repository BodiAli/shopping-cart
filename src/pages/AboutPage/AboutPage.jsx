import { useEffect } from "react";
import styles from "./AboutPage.module.css";

// Icons
import { FaGithub } from "react-icons/fa";
import { RiGitRepositoryFill } from "react-icons/ri";

function AboutPage() {
  useEffect(() => {
    document.title = "About | GameVault";

    return () => {
      document.title = "GameVault";
    };
  }, []);

  return (
    <>
      <main className={styles.main}>
        <div className={styles.titleContainer}>
          <h2 className={styles.title}>GameVault</h2>
        </div>
        <p>
          Welcome to GameVault, your ultimate destination for exploring the world of video games! At
          GameVault, we bring together a vast collection of games, offering you detailed description,
          screenshots, and information about your favorite titles.
        </p>
      </main>
      <footer className={styles.footer}>
        <p>Made by BodiAli</p>
        <div>
          <a href="https://github.com/BodiAli">
            <FaGithub color="#386c5f" />
          </a>
          <a href="https://github.com/BodiAli/shopping-cart">
            <RiGitRepositoryFill color="#386c5f" />
          </a>
        </div>

        <p>Â© {new Date().getFullYear()} GameVault. All rights reserved.</p>
      </footer>
    </>
  );
}

export default AboutPage;
