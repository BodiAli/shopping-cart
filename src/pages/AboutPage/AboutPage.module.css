.main {
  background-color: var(--green);
  padding: 2rem;
  display: flex;
  flex-direction: column;
  gap: 20px;
  min-height: 70vh;
}

.titleContainer {
  align-self: center;
  border-radius: 5px;
  padding: 1rem;
  background-color: var(--primary);
}

.title {
  font-family: "Merriweather", Times, serif;
  font-size: 2.5rem;
  font-weight: bold;
  background-image: url(/green-bg-text.jpg);
  background-clip: text;
  background-position: center;
  color: transparent;
  filter: brightness(1.8);
  text-align: center;
}

.main > p {
  font-size: 1.3rem;
  line-height: 1.3;
  margin-bottom: 200px;
}

.footer {
  background-color: var(--green);
  padding: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
  gap: 20px;
}

.footer > p:first-child {
  font-family: "Merriweather", Times, serif;
  font-size: 1.8rem;
}

.footer > div {
  font-size: 2rem;
}

.footer a:hover {
  filter: brightness(1.3);
}

.footer a:active {
  filter: brightness(0.8);
}

@media (max-width: 710px) {
  .main > p {
    margin-bottom: 100px;
  }
}
