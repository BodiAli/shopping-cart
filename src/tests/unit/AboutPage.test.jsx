import { render, screen } from "@testing-library/react";
import { createMemoryRouter, RouterProvider } from "react-router-dom";
import AboutPage from "../../pages/AboutPage/AboutPage";

const router = createMemoryRouter([
  {
    path: "/",
    element: <AboutPage />,
  },
]);

const wrapper = () => <RouterProvider router={router} />;

describe("AboutPage component", () => {
  test("should render correct elements", () => {
    const { container } = render(<AboutPage />, { wrapper });
    expect(container).toMatchSnapshot();
  });

  test("should change document title", () => {
    render(<AboutPage />, { wrapper });
    expect(document.title).toBe("About | GameVault");
  });

  test("should render website name", () => {
    render(<AboutPage />, { wrapper });
    expect(screen.getByRole("heading", { level: 2, name: /GameVault/ })).toBeInTheDocument();
  });

  test("should render a paragraph that describes the website", () => {
    render(<AboutPage />, { wrapper });
    expect(screen.getByText(/welcome to gamevault, your ultimate destination/i)).toBeInTheDocument();
  });

  test("should render author information", () => {
    render(<AboutPage />, { wrapper });
    const links = screen.getAllByRole("link");

    links.forEach((link) => {
      expect(link).toBeInTheDocument();
    });
    expect(screen.getByText(/made by bodiali/i)).toBeInTheDocument();
  });

  test("should render copyright reserved with the same year", () => {
    render(<AboutPage />, { wrapper });
    const currentDate = new Date();
    expect(
      screen.getByText(`Â© ${currentDate.getFullYear()} GameVault. All rights reserved.`)
    ).toBeInTheDocument();
  });
});
