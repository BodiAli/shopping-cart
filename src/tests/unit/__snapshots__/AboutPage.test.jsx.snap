// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AboutPage component > should render correct elements 1`] = `
<div>
  <main
    class="_main_45a761"
  >
    <div
      class="_titleContainer_45a761"
    >
      <h2
        class="_title_45a761"
      >
        GameVault
      </h2>
    </div>
    <p>
      Welcome to GameVault, your ultimate destination for exploring the world of video games! At GameVault, we bring together a vast collection of games, offering you detailed description, screenshots, and information about your favorite titles.
    </p>
  </main>
  <footer
    class="_footer_45a761"
  >
    <p>
      Made by BodiAli
    </p>
    <div>
      <a
        href="https://github.com/BodiAli"
        target="_blank"
      >
        <svg
          color="#386c5f"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(56, 108, 95);"
          viewBox="0 0 496 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
          />
        </svg>
      </a>
      <a
        href="https://github.com/BodiAli/shopping-cart"
        target="_blank"
      >
        <svg
          color="#386c5f"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          style="color: rgb(56, 108, 95);"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M13 21V23.5L10 21.5L7 23.5V21H6.5C4.567 21 3 19.433 3 17.5V5C3 3.34315 4.34315 2 6 2H20C20.5523 2 21 2.44772 21 3V20C21 20.5523 20.5523 21 20 21H13ZM7 19V17H13V19H19V16H6.5C5.67157 16 5 16.6716 5 17.5C5 18.3284 5.67157 19 6.5 19H7ZM7 5V7H9V5H7ZM7 8V10H9V8H7ZM7 11V13H9V11H7Z"
          />
        </svg>
      </a>
    </div>
    <p>
      Â© 
      2024
       GameVault. All rights reserved.
    </p>
  </footer>
</div>
`;
